fog/ledger/server/src/db_fetcher.rs

```diff
@@ -234,59 +237,60 @@ impl<
                     e
                 );
                 std::thread::sleep(Self::ERROR_RETRY_FREQUENCY);
-                *next_block_index
+                // We errored so we don't know how many blocks are in the ledger
+                return 0;
             }
-            Ok(BlockDataResponse {
-                result,
-                latest_block,
-            }) => {
-                // Tracing
-                let tracer = tracer!();
+            Ok(blocks) => blocks,
+        };
 
-                let mut span = block_span_builder(&tracer, "poll_block", *next_block_index)
-                    .with_start_time(start_time)
-                    .start(&tracer);
+        let latest_block = blocks.latest_block;
 
-                span.set_attribute(TELEMETRY_BLOCK_INDEX_KEY.i64(*next_block_index as i64));
+        if let Some(next_block) = blocks.results.get(0).and_then(|r| r.as_ref()) {
```

Comment:
This is a little hard to see in the GH ui
BeyondCompare seems to handle it nicer, 
![image](https://github.com/mobilecoinfoundation/mobilecoin/assets/101757619/94fcbe67-2ecc-4d15-b447-52c2026ac29a)